### ISSUE TITLE
onExit don't fire if we hover only the first pixel of the component

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1176]
The test fails on the commented code:
```
        window.sendMouseEvent(MouseEvent.MOUSE_PRESSED, x = 90, y = 50)
        window.sendMouseEvent(MouseEvent.MOUSE_RELEASED, x = 80, y = 50)
        awaitIdle()

        window.sendMouseEvent(MouseEvent.MOUSE_EXITED, x = 900, y = 500)
        awaitIdle()
        assertThat(onExits).isEqualTo(1) // fails here
```

Probably this is because we change pointerId on MOUSE_RELEASED (search `pointerId += 1` in DesktopOwners)

[The test](https://android-review.googlesource.com/c/platform/frameworks/support/+/1818155) is not merged yet

**Comment by igordmn** [comment]
The fix was reverted, as it causes another issues.

**Comment by igordmn** [comment]
The reproducer of the issue in a real application:

https://user-images.githubusercontent.com/5963351/142412664-54bf10c3-3a49-4e57-95d6-bff9448ee2f2.mp4

1. hover the first pixel of the scrollbar (not the second one)
2. press the mouse
3. release the mouse
4. move mouse to the left

Result: the scrollbar is still hovered

**Comment by akurasov** [comment]
Fixed in 1.0.0-rc2

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5265">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
